src/Parser/Mage/Frost/Modules/Features/IceLance.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import SpellIcon from 'common/SpellIcon';
+import { formatPercentage } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import EnemyInstances from 'Parser/Core/Modules/EnemyInstances';
+import AbilityTracker from 'Parser/Core/Modules/AbilityTracker';
+import Analyzer from 'Parser/Core/Analyzer';
+
+class IceLance extends Analyzer {
+	static dependencies = {
+		combatants: Combatants,
+		enemies: EnemyInstances,
+		abilityTracker: AbilityTracker,
+	}
+
+	iceLanceTargetID = 0;
+	hadFingersProc = 0;
+	nonShatteredCasts = 0;
+
+	on_byPlayer_cast(event) {
+		const spellId = event.ability.guid;
+		if (spellId !== SPELLS.ICE_LANCE_CAST.id) {
+			return;
+		}
+		this.hadFingersProc = 0;
+		this.iceLanceTargetID = event.targetID;
+		if (this.combatants.selected.hasBuff(SPELLS.FINGERS_OF_FROST.id)) {
+			this.hadFingersProc = 1;
+		}
+	}
+
+	on_byPlayer_damage(event) {
+		const spellId = event.ability.guid;
+		if (spellId !== SPELLS.ICE_LANCE_DAMAGE.id || event.targetID !== this.iceLanceTargetID) {
+			return;
+		}
+		const enemy = this.enemies.getEntity(event);
+		if (!enemy.hasBuff(SPELLS.WINTERS_CHILL.id) && this.hadFingersProc === 0) {
```

Comment:
Winter's Chill isn't the only debuff on target that can cause a Shatter. Also need to check for Water Elemental Freeze, Frost Nova, Ice Nova, Ring of Frost, and Glacial Spike (maybe more I'm not thinking of?). Probably should have an array of freeze effects to check against.

Reply:
I have all the IDs for these debuffs now, and i made the array ... but im not sure how to incorporate the array into hasBuff().

Reply:
nvm, i think i found the answer in the warlock modules :P